### PR TITLE
Support Role-Based Access Control II

### PR DESCRIPTION
**Pull request by SimFG**
#16986
rbac II
Signed-off-by: SimFG <bang.fu@zilliz.com>

### CODE REVIEW

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
run-cpu-e2e

File: internal/util/constant.go [hunk 1]
```diff
@@ -129,34 +63,60 @@ func StringList(stringMap map[string]struct{}) []string {
...
-	index, ok := commonpb.ObjectPrivilege_value[name]
+	_, ok := commonpb.ObjectPrivilege_value[name]
 	if !ok {
 		return ""
 	}
-	return ObjectPrivilegeAPI(index).String()
+	return MetaStore2API(name)
 }
 
 func PrivilegeNameForDb(name string) string {
```
Comment (jaime0815) [inline internal/util/constant.go:115]: PrivilegeNameForMetastore?

File: internal/proxy/privilege_interceptor.go [hunk 2]
```diff
@@ -87,17 +89,24 @@ func PrivilegeInterceptor(ctx context.Context, req interface{}) (context.Context
...
 	roleNames = append(roleNames, util.RolePublic)
-	resourceType := privilegeExt.ObjectType.String()
-	resourceNameIndex := privilegeExt.ObjectNameIndex
-	resourceName := funcutil.GetResourceName(req, privilegeExt.ObjectNameIndex)
-	resourcePrivilege := privilegeExt.ObjectPrivilege.String()
+	objectType := privilegeExt.ObjectType.String()
+	objectNameIndex := privilegeExt.ObjectNameIndex
+	objectName := funcutil.GetObjectName(req, objectNameIndex)
+	if checkCurUserObject(objectType, username, objectName) {
+		return ctx, nil
```
[the remark this answers is not in the document]
Comment (SimFG) [reply internal/proxy/privilege_interceptor.go:96]: The effect of `checkCurUserObject ` is to make user operate their own user information when the related privileges are not granted. like updating password or viewing your own role information.

**Review by congqixia — APPROVED** [review]
/lgtm

**Comment by SimFG** [comment]
@jaime0815 @congqixia Thanks your review :)

**Pull request merged**